velero-plugins/imagestream/restore.go

```diff
@@ -43,8 +43,8 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	itemMarshal, _ = json.Marshal(input.ItemFromBackup)
 	json.Unmarshal(itemMarshal, &imageStreamUnmodified)
 
-	skipImages := annotations[common.SkipImages]
-	if len(skipImages) != 0 {
+	skipImages := annotations[common.SkipImageCopy]
+	if len(skipImages) != 0 && skipImages == "true" {
 		p.Log.Info("Not running in OADP/CAM context, skipping copy of image.")
 		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
```

Comment:
Actually, here we probably need to distinguish between skipping image copy because we're not in MTC (so we'd want to restore the ImageStream as-is) and skipping image copy because MTC is configured for it, because in that case we need to restore an ImageStream with spec.Tags/status.Tags removed.
@dymurray I'm actually not so clear what we should do in the non-OADP/non-MTC case. Should we leave that as-is, or is it acceptable to restore a dummy ImageStream in that case as well? If we can always restore a dummy imagestream, then we can just remove tags right here. If we only want to restore a dummy imagestream when MTC sets skip-image-copy, then we probably need to make sure that mig-controller sets the annotation on the restore resource here, and we can distinguish between the two cases ( 1. non-MTC, skipping images because we can't copy them; 2. MTC, user specified skip-image-copy) by checking both annotations. If annotation is on ImageStream but not Restore, then restore item as-is (case 1), and if annotation is on both, then restore dummy ImageStream with tags removed (case 2)

Reply:
@sseago This change has been done.

Reply:
@seshapad I'm not sure I'm following the change here. Is the intent here to strip tags from images only when running from MTC (and for non-MTC/non-OADP we just restore the original ImageStream resource)? If that's the case, I'm not seeing anything here that would either 1) distinguish between these cases, or 2) strip tags from ImageStreams.